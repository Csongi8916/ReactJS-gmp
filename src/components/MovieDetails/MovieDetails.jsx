import React from 'react';
import Logo from '../Logo/Logo';

import classes from './MovieDetails.scss';

//import pic from '../../assets/testimg.jpg'

export default function MovieDetails() {
  return (
    <div className="MovieDetails">
      <div className="MovieDetails_Logo">
        <Logo />
      </div>
      <div className="MovieDetails_Container">
        <img src="https://image.tmdb.org/t/p/w300/ldoY4fTZkGISMidNw60GHoNdgP8.jpg" className="MovieDetails_Img" />
        <div className="MovieDetails_Info">
          <span className="MovieDetails_Title">Guardians of the Galaxy Vol. 3</span>
          <span className="MovieDetails_Rate">4.3</span>
          <p className="MovieDetails_Description">Pulp Fiction is a 1994 American[4] crime film written and directed by Quentin Tarantino, who conceived it with Roger Avary.[5] Starring John Travolta,
          Samuel L. Jackson, Bruce Willis, Tim Roth, Ving Rhames, and Uma Thurman, it tells several stories of criminal Los Angeles. The title refers to the pulp
          magazines and hardboiled crime novels popular during the mid-20th century, known for their graphic violence and punchy dialogue.</p>
        </div>
      </div>
    </div>
  )
}
