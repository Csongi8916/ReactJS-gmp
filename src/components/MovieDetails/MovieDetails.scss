$nrRedColor: #f65261;
$nrGrenColor: #a1e66f;
$nrBlackColor: #232323;
$nrWhiteColor: #ffffff;

.MovieDetails {
  display: flex;
  flex-direction: column;
  background-color: $nrBlackColor;
  padding-top: 20px;
  padding-bottom: 60px;

  .MovieDetails_Logo {
    margin-left: 50px;
  }

  .MovieDetails_Container {
    display: flex;
    margin: 40px 50px;

    .MovieDetails_Img {
    }

    .MovieDetails_Info {
      margin-left: 40px;
      color: #fff;
    }

    .MovieDetails_Title {
      font-size: 38px;
    }

    .MovieDetails_Rate {
      margin-left: 20px;
      padding: 15px;
      border: 2px solid #fff;
      border-radius: 50%;
      font-size: 25px;
      color: $nrGrenColor;
    }

    .MovieDetails_Description {
      margin-top: 45px;
      text-align: justify;
    }
  }
}


