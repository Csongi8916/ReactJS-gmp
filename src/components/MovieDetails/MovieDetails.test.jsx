import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import MovieDetails from './MovieDetails';

describe('MovieDetails Snapshpt tests:', () => {
  test('Should render correctly', () => {
    const tree = renderer.create(
      <MovieDetails
        title={'Test title'}
        image={'test img'}
        year={'2000-01-01'}
        rate={'5.5'}
        genres={['testg1', 'testg2']}
        runtime={'180'}
        overview={'Test description'}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});