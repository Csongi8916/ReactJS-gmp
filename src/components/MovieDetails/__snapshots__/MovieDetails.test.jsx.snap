// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails Snapshpt tests: Should render correctly 1`] = `
<div
  className="MovieDetails"
>
  <div
    className="MovieDetails_Header"
  >
    <div
      className="MovieDetails_Logo"
    >
      <span
        className="root"
      >
        <span
          className="main-part"
        >
          netflix
        </span>
        <span>
          rolulette
        </span>
      </span>
    </div>
    <a
      className="NxrSearchButton"
      href="#"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-search fa-w-16 fa-lg fa-rotate-90"
        data-icon="search"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </a>
  </div>
  <div
    className="MovieDetails_Container"
  >
    <img
      height="450"
      src="test img"
      width="330"
    />
    <div
      className="MovieDetails_Info"
    >
      <span
        className="MovieDetails_Title"
      >
        Test title
      </span>
      <span
        className="MovieDetails_Rate"
      >
        5.5
      </span>
      <p
        className="MovieDetails_Genre"
      >
        testg1 & testg2
      </p>
      <div
        className="MovieDetails_Time"
      >
        <span
          className="MovieDetails_DateTime"
        >
          2000
        </span>
        <span
          className="MovieDetails_DateTimeLabel"
        >
          year
        </span>
        <span
          className="MovieDetails_DateTime"
        >
          180
        </span>
        <span
          className="MovieDetails_DateTimeLabel"
        >
          min
        </span>
      </div>
      <p
        className="MovieDetails_Description"
      >
        Test description
      </p>
    </div>
  </div>
</div>
`;
